services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -978,8 +979,13 @@ public void deleteStore(
       AdminOperation message = new AdminOperation();
       message.operationType = AdminMessageType.DELETE_STORE.getValue();
       message.payloadUnion = deleteStore;
-
-      sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
+      // When multiple store deletions happened in a short time range, it is possible that the store deletion message
+      // will not be consumed in time, as previous store deletion message is still being processed.
+      try {
+        sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
```

Comment:
so.... the effect here is that we will wait up to a timeout, then if we go above that timeout, we'll log an error and then we'll return to the client like everything is fine (as opposed to barfing an error).

....At that point, why wait at all?  Why not fire and forget?

Reply:
Yeah, actually that's a good point. Even we waited until consumed still do not make too much sense.  Consumed does not mean admin execution successfully.